index.js

```diff
@@ -0,0 +1,251 @@
+      }
+    }
+
+    function getChangesPercentage(data) {
+      let aTag = document.getElementById("container").querySelectorAll(".aTag");
+      for (let i = 0; i < data.length; i++) {
+        let div = document.createElement("div");
+        div.className = "changesPercentage";
+        if (data[i].profile.changesPercentage < 0) {
+          div.innerHTML = `${data[i].profile.changesPercentage}%`;
+          div.classList.add("red");
+        } else {
+          div.innerHTML = `+${data[i].profile.changesPercentage}%`;
+          div.classList.add("green");
+        }
+        aTag[i].appendChild(div);
+      }
+    }
+
+    function fetchNTimes() {
+      let aTag = document.getElementById("container").querySelectorAll(".aTag");
+      let symbolArray = [];
+      aTag.forEach((element) =>
+        symbolArray.push(element.innerHTML.split("(")[1].split(")")[0])
+      );
+      let responses = [];
+      for (let i = 0; i < symbolArray.length; i++) {
+        let symbol = symbolArray[i];
+        let response = fetch(
+          `https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3/company/profile/${symbol}`
+        );
+
+        responses.push(response);
+      }
+      return Promise.all(responses);
+    }
+    const awaitJson = (responses) =>
+      Promise.all(
+        responses.map((response) => {
+          if (response.ok) return response.json();
+          throw new Error(response.statusText);
+        })
+      );
+
+    function displaySearch(data) {
+      let results = document.getElementById("display-results");
+      if (results.firstElementChild) {
+        results.removeChild(results.firstElementChild);
+      }
+
+      let listLength = Math.min(10, data.length);
+      let container = document.createElement("div");
+      container.style.height = `${74 * listLength}px`;
+      container.className = "container";
+      container.id = "container";
+      for (let i = 0; i < listLength; i++) {
+        let element = `${data[i].name}  (${data[i].symbol})`;
+        let aTag = document.createElement("a");
+        let para = document.createElement("p");
+        aTag.className = `aTag`;
+        aTag.href = `company.html?symbol=${data[i].symbol}`;
+        aTag.innerHTML = element;
+        para.appendChild(aTag);
+        container.appendChild(para);
+      }
+      results.appendChild(container);
+    }
+
+    function initSearch() {
+      let button = document.querySelector("button");
+      button.addEventListener("click", dataFetch);
+    }
+
+    initSearch();
+  }
+
+  function showLoadingScreen() {
+    let loading = document.createElement("div");
+    loading.innerHTML = `Loading...`;
+    let displayStock = document.getElementById("loadDiv");
+    displayStock.appendChild(loading);
+    return loading;
+  }
+
+  function displayCompanyPage() {
+    function catSymbol() {
+      let urlParams = new URLSearchParams(window.location.search);
+      queryString = urlParams.toString();
```

Comment:
There are better ways to extract data from the URLSearchParams object, see here:

https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#examples